source/game/system/map/MapdataCheckPoint.cc

```diff
@@ -95,15 +95,120 @@ void MapdataCheckPoint::initCheckpointLinks(MapdataCheckPointAccessor &accessor,
     }
 }
 
+/// @addr{0x80510D7C}
+/// @see MapdataCheckPoint::checkSectorAndDistanceRatio
+/// @rename
+MapdataCheckPoint::SectorOccupancy MapdataCheckPoint::checkSectorAndDistanceRatio(
+        const EGG::Vector3f &pos, f32 &completion) const {
+    EGG::Vector2f p1 = m_right;
+    p1.y = pos.z - p1.y;
+    p1.x = pos.x - p1.x;
+
+    for (auto &next : m_nextPoints) {
```

Comment:
```suggestion
    for (size_t i = 0; i < m_nextCount; ++i) {
        MapdataCheckPoint *next = m_nextPoints[i];
```
We may only iterate over the initialized next checkpoints.